src/Tizen.NUI/src/public/WindowSystem/QuickPanelClient.cs

```diff
@@ -0,0 +1,422 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Tizen.Applications.Exceptions;
+
+namespace Tizen.NUI.WindowSystem.Shell
+{
+    /// <summary>
+    /// Class for the Tizen quickpanel client.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public class QuickPanelClient : IDisposable
+    {
+        private TizenShell _tzsh;
+        private IntPtr _tzshQpClient;
+        private int _tzshWin;
+        private bool disposed = false;
+
+        private int _visibleEventType;
+        private IntPtr _visibleEventHandler;
+        private Interop.QuickPanelClient.QuickPanelEventCallback _onVisibleChanged;
+        private event EventHandler<StateVisible> _visibleChanged;
+        /// <summary>
+        /// Emits the event when the visible is changed.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public event EventHandler<StateVisible> VisibleChanged
+        {
+            add
+            {
+                if (_visibleChanged == null)
+                {
+                    if (_visibleEventType == 0)
+                    {
+                        _visibleEventType = Interop.TizenShell.EventTypeNew(_tzsh.GetNativeHandle(), Interop.QuickPanelClient.EventStringVisible);
+                    }
+                    _onVisibleChanged = OnVisibleChanged;
+                    _visibleEventHandler = Interop.QuickPanelClient.EventHandlerAdd(_tzshQpClient, _visibleEventType, _onVisibleChanged, IntPtr.Zero);
+                    if (_visibleEventHandler == IntPtr.Zero)
+                    {
+                        int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                        _tzsh.ErrorCodeThrow(err);
+                    }
+                }
+                _visibleChanged += value;
+            }
+            remove
+            {
+                int _res;
+                _visibleChanged -= value;
+                if (_visibleChanged == null)
+                {
+                    _res = Interop.QuickPanelClient.EventHandlerDel(_tzshQpClient, _visibleEventHandler);
+                    _tzsh.ErrorCodeThrow(_res);
+                }
+            }
+        }
+
+        private int _orientationEventType;
+        private IntPtr _orientationEventHandler;
+        private Interop.QuickPanelClient.QuickPanelEventCallback _onOrientationChanged;
+        private event EventHandler<StateOrientation> _orientationChanged;
+        /// <summary>
+        /// Emits the event when the orientation is changed.
```

Comment:
orientation of QuickPanel ?
(As I know, QuickPanel and QuickPanelClient are different. We need to specify if the orientation is not for QuickPanelClient)

Reply:
I fix it to
Emits the event when the orientation of the quickpanel service window is changed.